File: ios/MiniappOptions.swift
//
//  MiniappOptions.swift
//  ExpoBoxoSdk
//
//  Created by Azamat Kushmanov on 18/9/24.
//

import ExpoModulesCore

struct MiniappOptions: Record {
    @Field
    var appId: String = ""
    
    @Field
    var data: [String : Any]?
    
    @Field
    var theme: String?
    
    @Field
    var extraUrlParams: [String : String]?
    
    @Field
    var urlSuffix: String?
    
    @Field
    var colors: [String : String]?
    
    @Field
    var enableSplash: Bool = true
    
    @Field
    var saveState: Bool = true
}

File: android/src/main/java/io/boxo/expo/MiniappOptions.kt
package io.boxo.expo

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class MiniappOptions : Record {
    @Field
    val appId: String = ""

    @Field
    val data: Map<String, Any>? = null

    @Field
    val theme: String? = null

    @Field
    val extraUrlParams: Map<String, Any>? = null

    @Field
    val urlSuffix: String? = null

    @Field
    val colors: Map<String, String>? = null

    @Field
    val enableSplash: Boolean = true

    @Field
    val saveState: Boolean = false

    @Field
    val pageAnimation: String? = null
}